### PR TITLE
Update register fragment unit tests

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Additional tests for https://github.com/opensrp/fhircore/pull/2244

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [x] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt [hunk 1]
```diff
@@ -90,6 +87,13 @@ class PatientExtensionTest : RobolectricTest() {
     Assert.assertEquals(expectedAge14, calculateAge(getDateFromDaysAgo(1826), context))
   }
 
+  @Ignore("Test is flaky and period dependent - it(+ Code) should be refactored")
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:90]: Could we add a ticket to track this as _Tech Debt_?

Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:90]: Actually it is related to this ticket. You can add it here https://github.com/opensrp/fhircore/issues/2184

Comment (Rkareko) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:90]: Ticket created [here](https://github.com/opensrp/fhircore/issues/2260)

Comment (Rkareko) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:90]: closed the new ticket and linked it to the earlier one

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt [hunk 2]
```diff
@@ -69,7 +69,7 @@ class RegisterFragment : Fragment(), OnSyncListener, Observer<QuestionnaireSubmi
 
   @Inject lateinit var syncListenerManager: SyncListenerManager
 
-  private val appMainViewModel by activityViewModels<AppMainViewModel>()
+  val appMainViewModel by activityViewModels<AppMainViewModel>()
```
Comment (ndegwamartin) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:72]: `protected` visibility could also have worked here if the reason was for using it in the **RegisterFragmentTest** class. Additional annotation with `@VisibileForTesting` would be good

**Review by ndegwamartin — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**